### PR TITLE
Add compatibility with old macOS versions

### LINKED ISSUE #732: 1.0.1 fails to build on macosx-10.9-x86_64-3.8 due to kFSEventStreamEventFlagItemCloned
kFSEventStreamEventFlagItemCloned was added in macOS 10.13. Builds for older OS releases should wrap unsupported symbols for backward compatibility even at the loss of minor functionality.

```
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/xxx/anaconda3/include -arch x86_64 -I/Users/xxx/anaconda3/include -arch x86_64 -DWATCHDOG_VERSION_STRING="1.0.1" -DWATCHDOG_VERSION_MAJOR=1 -DWATCHDOG_VERSION_MINOR=0 -DWATCHDOG_VERSION_BUILD=1 -I/Users/xxx/proj/macrotomic/play/venv/include -I/Users/xxx/anaconda/include/python3.8 -c src/watchdog_fsevents.c -o build/temp.macosx-10.9-x86_64-3.8/src/watchdog_fsevents.o -std=c99 -pedantic -Wall -Wextra -fPIC -Wno-nullability-completeness -Wno-nullability-extension -Wno-newline-eof -Wno-error=unused-command-line-argument
    src/watchdog_fsevents.c:160:25: error: use of undeclared identifier 'kFSEventStreamEventFlagItemCloned'
    FLAG_PROPERTY(IsCloned, kFSEventStreamEventFlagItemCloned)
                            ^
    src/watchdog_fsevents.c:272:63: warning: unused parameter 'event_ids' [-Wunused-parameter]
                                   const FSEventStreamEventId     event_ids[])
```

### PR DESCRIPTION
**Pull request by CCP-Aporia**
Fix #732 

Note that this is mostly untested as I don't have access to old machines and all my local python versions (installed via MacPorts) were configured for Big Sur which makes it impossible to build the extension for an earlier version.

### CODE REVIEW

**Comment by BoboTiG** [comment]
Should we add this code before that line: https://github.com/gorakhargosh/watchdog/blob/44b9618f147bb74bbfab382964d4e6ae674e1a15/setup.py#L76
```python

                # macOS 10.13 is the oldest version supported by Apple.
                '-mmacosx-version-min={}'.format(os.environ.get('MACOSX_DEPLOYMENT_TARGET', '10.13'),
```
?

Maybe 10.13 is to high for now and we could stick with 10.9 🤔

**Comment by CCP-Aporia** [comment]
That's something I tried, but unfortunately it doesn't work with all Python versions on macOS. The versions I have installed via MacPorts have Python compiled with `MACOSX_DEPLOYMENT_TARGET=11.0` which causes a compile error when trying to build a C extensions with a version lower than `11.0`. See this example (added `setenv = MACOSX_DEPLOYMENT_TARGET=10.8` to `tox.ini`):

```
running build_ext
    building '_watchdog_fsevents' extension
    creating build/temp.macosx-11.0-x86_64-3.9
    creating build/temp.macosx-11.0-x86_64-3.9/src
    /usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -DWATCHDOG_VERSION_STRING="1.0.2" -DWATCHDOG_VERSION_MAJOR=1 -DWATCHDOG_VERSION_MINOR=0 -DWATCHDOG_VERSION_BUILD=2 -I/Users/thomas/github.com/CCP-Aporia/watchdog/.tox/py39/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c src/watchdog_fsevents.c -o build/temp.macosx-11.0-x86_64-3.9/src/watchdog_fsevents.o -std=c99 -pedantic -Wall -Wextra -fPIC -Wno-nullability-completeness -Wno-nullability-extension -Wno-newline-eof -Wno-error=unused-command-line-argument
    error: $MACOSX_DEPLOYMENT_TARGET mismatch: now "10.8" but "11.0" during configure
```

**Comment by BoboTiG** [comment]
Well it seems safer to not alter that part of the code then :)

**Comment by BoboTiG** [comment]
Could you add the changelog entry please? I will merge right after.

**Review by BoboTiG — APPROVED** [review]

**Comment by BoboTiG** [comment]
Thanks a lot @CCP-Aporia :)

**Pull request merged**
